### ISSUE TITLE
The meal_price_outlier_classifier classifier is broken

### ISSUE THREAD

**Issue by sergiomario** [okfn-brasil/serenata-de-amor#506]
**What is the problem?**
There is an error executing the script `pipenv run ansible-playbook update.yml`.
This script runs weekly to update the *Câmara dos Deputados* data and classify suspicious invoices. As defined in the [data update procedure](https://github.com/okfn-brasil/serenata-de-amor/tree/master/contrib/update).
![erro_rosie](https://user-images.githubusercontent.com/2698516/70950658-01e4f780-2040-11ea-83ec-a1d690fb3fca.png)

```
File serenata-de-amor/rosie/rosie/chamber_of_deputies/classifiers/meal_price_outlier_classifier.py, line 105, in __normalize_string

nfkd_form = unicodedata.normalize('NFKD', string.lower())", 
"AttributeError: 'float' object has no attribute 'lower'"
```

**How can this be addressed?**
Type validation can be done by ensuring that a String object is being manipulated.

**Comment by cuducos** [comment]
In the Chamber of Deputies [adapter, we only enforce a couple of fields to be read as strings](https://github.com/okfn-brasil/serenata-de-amor/blob/55b07726a49e9ff7149b313f2c13d6199ab2f4a2/rosie/rosie/chamber_of_deputies/adapter.py#L25-L30):

```python
    DTYPE = {
        'applicant_id': np.str,
        'cnpj_cpf': np.str,
        'congressperson_id': np.str,
        'subquota_number': np.str
    }
```

Probably because in most cases Pandas would read the column `recipient` as strings as well. The error suggests it doesn't (the _traceback_ leads to a single call of the `__normalize_string` method with is called using the dataset `recipient` column).

Also, [the adapter creates the `recipient` column from the `supplier` column](https://github.com/okfn-brasil/serenata-de-amor/blob/55b07726a49e9ff7149b313f2c13d6199ab2f4a2/rosie/rosie/chamber_of_deputies/adapter.py#L23). So, my best guess to try to fix this bug would be to enforce the `np.str` as the `dtype` of the supplier, adding a single line to block I pasted above:

```python
    DTYPE = {
        'applicant_id': np.str,
        'cnpj_cpf': np.str,
        'congressperson_id': np.str,
        'subquota_number': np.str,
        'supplier': np.str,
    }
```

**However** it's extremely unlikely that Pandas is reading this column as a numeric one for no reason. We should make sure the CSV inputs still consistent to whatever we're expecting from them.

**Comment by luizfzs** [comment]
I wanted to give it a try fixing this problem but I'm having a hard time to start Rosie, since it requires a lot of memory. I'll keep working on get it running tonight, but don't hesitate (anyone) on working on this issue too, since it might take some time for me to get it running.

**Comment by luizfzs** [comment]
I was able to make Rosie work on my machine with some Macgyver. 
I'm able to reproduce the issue using only the reimbursements from 2019. I followed @cuducos approach to enforce the `dtype` for `supplier` on the `adapter`, still to no avail.
Since the issue is happening on `meal_price_outlier_classifier.py`, I added a `print` command to `__normalize_string` so it prints the value when **it's not a string**. Here's the result:
`type:<class 'float'> ; val: nan`

From my check, some suppliers are actually empty but shouldn't cause the issue.
